entrypoint.sh

```diff
@@ -1,12 +1,26 @@
 #!/bin/bash
 
-set -e
+# Function to URL-encode a string
+urlencode() {
+  local string="$1"
+  local encoded=""
+  local length="${#string}"
+  for ((i = 0; i < length; i++)); do
+    local char="${string:i:1}"
+    case "$char" in
+    [a-zA-Z0-9.~_-]) encoded+="$char" ;;
+    *) encoded+="%"$(printf "%02X" "'$char") ;;
+    esac
+  done
+  echo "$encoded"
+}
 
 # This supports both the old environment variables and new ones specific to the Figma setup.
 if [ -n "$PG_FIGMA_FOR_JIRA_DB_ROLE" ]; then
-  DATABASE_URL=postgresql://${PG_FIGMA_FOR_JIRA_DB_ROLE}:${PG_FIGMA_FOR_JIRA_DB_PASSWORD}@${PG_FIGMA_FOR_JIRA_DB_HOST}:${PG_FIGMA_FOR_JIRA_DB_PORT}/${PG_FIGMA_FOR_JIRA_DB_SCHEMA}
+  DATABASE_URL="postgresql://${PG_FIGMA_FOR_JIRA_DB_ROLE}:${PG_FIGMA_FOR_JIRA_DB_PASSWORD}@${PG_FIGMA_FOR_JIRA_DB_HOST}:${PG_FIGMA_FOR_JIRA_DB_PORT}/${PG_FIGMA_FOR_JIRA_DB_SCHEMA}"
 else
-  DATABASE_URL=postgresql://${PG_INTEGRATIONS_JIRA_USER}:${PG_INTEGRATIONS_JIRA_PASSWORD}@${PG_INTEGRATIONS_HOST}:${PG_INTEGRATIONS_PORT}/${PG_INTEGRATIONS_DATABASE}
+  ENCODED_PASSWORD="$(urlencode "$PG_INTEGRATIONS_JIRA_PASSWORD")"
```

Comment:
**suggestion:** Please, consider encoding other components as well.

According to the [Prisma docs](https://www.prisma.io/docs/reference/database-reference/connection-urls#special-characters):

> For MySQL, PostgreSQL and CockroachDB you must [percentage-encode special characters](https://developer.mozilla.org/en-US/docs/Glossary/percent-encoding) in **any part** of your connection URL - including passwords.

Reply:
That suggestion makes sense to me! I think in our case, we don't have any special characters in the other configuration strings, and they're unlikely to change. If there are no special characters, then urlencode is a no-op, so I think this is alright as-is for now.

```
Elaine $ foo="ABC"
Elaine $ bar="$(urlencode $foo)"
Elaine $ echo "$bar"
ABC
```